controllers/chat/sendAnonymousMessage.js

```diff
@@ -101,7 +109,43 @@ const sendAnonymousMessage = async (req, res) => {
       });
     }
 
-    const chat = await channel.sendMessage(baseMessage);
+    const chat = await channel.sendMessage(baseMessage, {skip_push: true});
+    let notificationPayload = {
+      title: `${senderInfo?.user?.username}`,
+      body: `${message.substring(0, 100)}`
+    };
+    let dataPayload = {
+      channel_id: channelId,
+      user_id: req.userId,
+      messages_id: chat.message.id,
+      message,
+      message_schema: currentMessageType,
+      type: 'message.new',
+      status: 'sent',
+      is_big_message: 'false',
+      attachment: JSON.stringify(attachments) ?? '',
+      created_at: chat.message.created_at,
+      is_annoymous: 'false'
+    };
+    const dataSizeInBytes = Buffer.byteLength(JSON.stringify(dataPayload), 'utf8');
+    const dataSizeInKilobytes = dataSizeInBytes / 1024;
+    if (dataSizeInKilobytes > 4) {
+      dataPayload.is_big_message = 'true';
+      dataPayload.attachment = '';
+    }
+    channelMember.forEach((member) => {
+      let payload = {
+        notification: notificationPayload,
+        data: {
+          ...dataPayload,
+          is_annoymous: member.is_anonymous.toString()
+        }
+      };
+      BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload, {
+        priority: 'high',
+        content_available: true
+      });
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code seems to be handling the creation of a chat message and sending push notifications. It would be better to separate these two concerns into different functions for better modularity and readability. For example, you could have one function `createChatMessage()` and another `sendPushNotifications()`. 

Also, there's no error handling around the `BetterSocialCore.fcmToken.sendChatNotification()` call. What happens if sending a notification fails? You might want to add some error handling here to ensure that your application can gracefully handle such situations.

Lastly, the `is_big_message` field is a string, but it seems like it should be a boolean. Consider changing its type to boolean for more accurate representation.
```diff
-      dataPayload.is_big_message = 'true';
+      dataPayload.is_big_message = true;
```
And similarly,
```diff
-      dataPayload.is_big_message = 'false';
+      dataPayload.is_big_message = false;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->